lib/macho/macho_file.rb

```diff
@@ -233,10 +264,12 @@ def dylib_id=(new_id)
       raise ArgumentError.new("new ID must be a String") unless new_id.is_a?(String)
       return unless dylib?
 
-      dylib_cmd = command(:LC_ID_DYLIB).first
-      old_id = dylib_id
+      old_cmd = command(:LC_ID_DYLIB).first
+      new_cmd = LoadCommand.create(:LC_ID_DYLIB, new_id, old_cmd.timestamp,
+        old_cmd.current_version, old_cmd.compatibility_version)
```

Comment:
Creating one if it's missing is probably not a good idea, particularly since we wouldn't know to what values we should set `timestamp`, `current_version`, and `compatibility_version`.

But raising an exception if there's none sounds like a reasonable thing to do and would avoid a hard to interpret failure when trying to access the above mentioned fields of `nil`.